jax/_src/lax/control_flow/loops.py

```diff
@@ -1962,12 +1962,19 @@ def fori_loop(lower, upper, body_fun, init_val):
   if use_scan:
     if unroll is None:
       unroll = False
-    if config.disable_jit.value and upper_ == lower_:
-      # non-jit implementation of scan does not support length=0
+    if (length := upper_ - lower_) < 0 or (
```

Comment:
I find it confusing to define `length` in the `if` statement given that it's also used in the outer scope further below. I'd just put the definition on its own line.

Reply:
Sure, done.